src/browser/BrowserMonitoringService.ts

```diff
@@ -0,0 +1,78 @@
+import { datadogLogs } from '@datadog/browser-logs';
+
+import { Logger } from '../shared/Logger';
+import { ConsoleLogger } from '../shared/ConsoleLogger';
+
+const ddLogger = datadogLogs.logger;
+
+interface RemoteMonitoringServiceParams {
+  serviceName: string;
+  serviceVersion: string;
+  serviceEnv: string;
+  clientToken: string;
+}
+
+export class BrowserMonitoringService {
+  private logger: Logger;
+  private remoteMonitoringServiceParams:
+    | RemoteMonitoringServiceParams
+    | undefined;
+
+  constructor(remoteMonitoringServiceParams?: RemoteMonitoringServiceParams) {
+    const { serviceName, serviceVersion, serviceEnv, clientToken } =
+      remoteMonitoringServiceParams ?? {};
+
+    const isSomeRemoteMonitoringServiceParamMissing =
+      !serviceName || !serviceVersion || !serviceEnv || !clientToken;
+
+    if (!remoteMonitoringServiceParams) {
+      this.logger = new ConsoleLogger();
+    } else if (
+      remoteMonitoringServiceParams &&
+      isSomeRemoteMonitoringServiceParamMissing
+    ) {
+      this.logger = new ConsoleLogger();
+      this.logger.warn(
+        '[MonitoringService] can not initialize remote monitoring service, because some variables are missing. Initializing with console instead',
+      );
+      this.logger.warn(`
+        serviceName - ${serviceName}
+        serviceVersion - ${serviceVersion}
+        serviceEnv - ${serviceEnv}
+        clientToken - ${clientToken}
+      `);
+    } else {
+      this.logger = ddLogger;
+    }
+
+    this.remoteMonitoringServiceParams = remoteMonitoringServiceParams;
+  }
+
+  setupReportingNativeLogs() {
+    if (!this.remoteMonitoringServiceParams) {
+      this.reportError(
+        new Error('Can not setup remote monitoring without all needed params'),
+      );
+
+      return;
+    }
+
+    datadogLogs.init({
+      clientToken: this.remoteMonitoringServiceParams.clientToken,
+      service: this.remoteMonitoringServiceParams.serviceName,
+      version: this.remoteMonitoringServiceParams.serviceVersion,
+      env: this.remoteMonitoringServiceParams.serviceEnv,
+      // TODO: think about below
+      forwardErrorsToLogs: true,
+      sessionSampleRate: 100,
+    });
+  }
+
+  info(...args: [message: string, context?: object]) {
+    this.logger.info(...args);
+  }
+
```

Comment:
Nice idea, though about this. 
But I think last time we talked we agreed to expose only needed methods. 
And for now I wanted to start simple. 
Like with general log function we provide user with lots of methods (namely error, warn, info, debug, log) and I can not see now any need in those, since we don't yet have proper system when and how to use all of them correctly. 

And also reason why I explicitly extracted `error` from log level is that errors need to be handled, captured, reported and action need to happen as a result, while logs just need to be reported. So from consumer perspective - they are not similar. And actually it become obvious when you see that API is different, since we don't need to provide error argument for any other method (warn, log etc...). So extending those methods to accept Error param is kind of bloating API unnecessary for me.

Interesting what do you think about it?